core/tauri/src/updater/mod.rs

```diff
@@ -395,6 +402,28 @@ impl<R: Runtime> UpdateResponse<R> {
     self.update.body.as_ref()
   }
 
+  /// Add a `Header` to the update request.
+  pub fn header<K, V>(mut self, key: K, value: V) -> Result<Self>
+  where
+    HeaderName: TryFrom<K>,
+    <HeaderName as TryFrom<K>>::Error: Into<http::Error>,
+    HeaderValue: TryFrom<V>,
+    <HeaderValue as TryFrom<V>>::Error: Into<http::Error>,
+  {
+    self.update = self.update.header(key, value)?;
+    Ok(self)
+  }
+
+  /// Removes a header from the update request.
+  pub fn remove_header<K>(mut self, key: K) -> Result<Self>
+  where
+    HeaderName: TryFrom<K>,
+    <HeaderName as TryFrom<K>>::Error: Into<http::Error>,
+  {
+    self.update = self.update.remove_header(key)?;
+    Ok(self)
+  }
```

Comment:
would you explain why would you need this?

Reply:
probably because he needs a header for the JSON fetch request, but not for the install request

Reply:
Yes, exactly. If we set signed headers for the fetch request, they are not valid for the subsequent install request to S3 and S3 rejects them. Apologies for the delayed response.